packages/plugin-table/src/TableCellNode.ts

```diff
@@ -150,4 +120,58 @@ export class TableCellNode extends VNode {
     isActive(): boolean {
         return !this.managerCell;
     }
+    /**
+     * Set the given cell as manager of this cell. If null is passed, this cell
+     * will have no manager.
+     *
+     * @override
+     */
+    mergeWith(newManager: VNode): void {
+        const thisTable = this.ancestor(TableNode);
+        const otherTable = newManager.ancestor(TableNode);
+        if (!newManager.is(TableCellNode) || thisTable !== otherTable) return;
+
+        this.__managerCell = newManager;
+
+        // If we just set a new manager for this cell, also add this cell to
+        // the new manager's managed cells.
+        newManager.__managedCells.add(this);
+
+        // Copy the manager.
+        this.attributes = { ...newManager.attributes };
+        this.header = newManager.header;
+
+        // Move the children to the manager.
+        if (this.hasChildren()) {
+            newManager.append(new LineBreakNode());
+        }
+        newManager.append(...this.children);
+
+        // Hand the managed cells over to the manager.
+        for (const managedCell of this.managedCells) {
+            managedCell.mergeWith(newManager);
+            this.__managedCells.delete(managedCell);
+        }
+
+        // Copy the manager's row if an entire row was merged
+        const row = this.ancestor(TableRowNode);
+        if (row) {
+            const areAllManagedByManager = row
+                .children(TableCellNode)
+                .every(cell => cell.managerCell === newManager);
+            if (newManager.rowIndex !== this.rowIndex && areAllManagedByManager) {
+                const managerRow = this.managerCell.ancestor(TableRowNode);
+                row.header = managerRow.header;
+                row.attributes = { ...managerRow.attributes };
+            }
+        }
+    }
+    unmerge(): void {
+        if (this.__managerCell) {
+            // If we just removed this cell's manager, also remove this cell
+            // from the old manager's managed cells.
+            this.__managerCell.__managedCells.delete(this);
```

Comment:
Accessing private properties here as well.

Reply:
Same comment as above. Those private properties are built to be handled locally by merge/unmerge.

Reply:
Ok so here `__managedCells` is not accessed by `this`. but I believe my above comment still apply though. Whaddayathink?